Dockerfile

```diff
@@ -1,32 +1,70 @@
-ARG EXPERIMENTS_VERSION
+FROM faasm/cpp-sysroot:0.0.26
 
-# We pin to a specific sysroot release
-FROM faasm/cpp-sysroot:0.0.22 as build-step
+# Download and install OpenMPI
+WORKDIR /tmp
+RUN wget https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-4.1.0.tar.bz2
+RUN tar xf openmpi-4.1.0.tar.bz2
+WORKDIR /tmp/openmpi-4.1.0
+RUN ./configure --prefix=/usr/local
+RUN make -j `nproc`
+RUN make install
+
+# The previous steps take a lot of time, so be careful not to invalidate the
+# Docker cache
+
+# -------------------------------
+# NATIVE MPI SETUP
+# -------------------------------
+
+# Add an mpirun user
+RUN adduser --disabled-password --gecos "" mpirun
+RUN echo "mpirun ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+
+# Set up SSH (for native MPI)
+RUN apt install -y openssh-server
+RUN mkdir /var/run/sshd
+RUN echo 'root:mpirun' | chpasswd
+
+COPY ./ssh/sshd_config /etc/ssh/sshd_config
+
+# Generate a key to be used by all hosts
+WORKDIR /home/mpirun/.ssh
+RUN ssh-keygen -b 2048 -t rsa -f /home/mpirun/.ssh/id_rsa -q -N ""
+
+# Copy into authorized keys
+RUN cp id_rsa.pub authorized_keys
+
+# Copy SSH config into place
+COPY ssh/config config
+
+# Set up perms on SSH files
+WORKDIR /home/mpirun
+RUN chmod -R 600 .ssh
+RUN chmod 700 .ssh
+RUN chmod 644 .ssh/id_rsa.pub
+RUN chown -R mpirun:mpirun .ssh
```

Comment:
@csegarragonz in the old version [here](https://github.com/faasm/experiment-base/blob/master/docker/experiment-base.dockerfile#L35) it looks like it was copying in your SSH keys too, why was this? If we need to access the containers ourselves we can use `kubectl`.

Reply:
As far as I can remember, MPI requires nodes to be able to ssh password-less between each others, so I created a key pair for deployments and copied into the image. 

However, it is way cleaner to generate it at container build time as we do now.